core/transientstore/store.go

```diff
@@ -95,7 +135,179 @@ func (provider *storeProvider) OpenStore(ledgerID string) (*Store, error) {
 
 // Close closes the TransientStoreProvider
 func (provider *storeProvider) Close() {
-	provider.dbProvider.Close()
+	if provider.dbProvider != nil {
+		provider.dbProvider.Close()
+	}
+	if provider.fileLock != nil {
+		provider.fileLock.Unlock()
+	}
+}
+
+// DeleteStore deletes the TransientStore for a given ledger.
+func (provider *storeProvider) DeleteStore(ledgerID string) error {
+	return provider.dbProvider.Drop(ledgerID)
+}
```

Comment:
This seems unnecessarily exported and adding to the interface and causing the mock regenration. I figured that you use this in test. For that you can simply cast to type `*storeProvider` in the test.

Reply:
I agree 110%